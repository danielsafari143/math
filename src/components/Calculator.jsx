import { useState } from 'react';
import calculate from './logic/calculate';

function Calculator() {
  const [btn, setBtn] = useState('');
  const [obj, setObj] = useState({});
  const [calc, setCalc] = useState({});

  const handleButtonClick = (e) => {
    let valeur = '';
    setBtn((p) => {
      valeur = p + e.target.value;
      setObj({ ...obj, next: parseInt(valeur, 10) });
      return valeur;
    });
  };

  const handleSigns = (e) => {
    setObj({ ...obj, total: parseInt(btn, 10), operation: e.target.value });
    setBtn('');
  };

  const handleEqual = (e) => {
    setCalc(() => ({ ...calc, obj, buttonName: e.target.value }));
    if (e.target.value === '=') {
      if (calc.obj !== undefined) {
        const num = calculate(calc.obj, calc.buttonName);
        setBtn(num.total);
        setCalc({});
      }
    } else if (e.target.value === '+/-') {
      if (calc.obj !== undefined) {
        const num = calculate(calc.obj, calc.buttonName);
        setBtn(num.next);
        setCalc({});
      }
    } else {
      setBtn('');
      setObj({});
      setCalc({});
    }
  };

  const handleInput = (e) => {
    setBtn(e.target.value);
  };

  return (
    <div className="calc">
      <input value={btn} onChange={handleInput} className="inpute" />
      <input type="button" onClick={handleEqual} className="signs gray" value="A/C" />
      <input type="button" onClick={handleEqual} className="signs gray" value="+/-" />
      <input type="button" onClick={handleSigns} className="signs gray" value="%" />
      <input type="button" onClick={handleSigns} className="signs orange" value="รท" />
      <input type="button" onClick={handleButtonClick} className="signs gray" value="7" />
      <input type="button" onClick={handleButtonClick} className="signs gray" value="8" />
      <input type="button" onClick={handleButtonClick} className="signs gray" value="9" />
      <input type="button" onClick={handleSigns} className="signs orange" value="x" />
      <input type="button" onClick={handleButtonClick} className="signs gray" value="4" />
      <input type="button" onClick={handleButtonClick} className="signs gray" value="5" />
      <input type="button" onClick={handleButtonClick} className="signs gray" value="6" />
      <input type="button" onClick={handleSigns} className="signs orange" value="-" />
      <input type="button" onClick={handleButtonClick} className="signs gray" value="1" />
      <input type="button" onClick={handleButtonClick} className="signs gray" value="2" />
      <input type="button" onClick={handleButtonClick} className="signs gray" value="3" />
      <input type="button" onClick={handleSigns} className="signs orange" value="+" />
      <input type="button" onClick={handleButtonClick} className="egals gray" value="0" />
      <input type="button" onClick={handleButtonClick} className="signs gray" value="." />
      <input type="button" onClick={handleEqual} className="signs orange" value="=" />
    </div>
  );
}

export default Calculator;
