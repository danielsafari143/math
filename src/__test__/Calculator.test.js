import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Confirming the display of calculator', () => {
  beforeAll(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  it('calculator displays', () => {
    const calculatorDisplay = render(<Calculator />);
    expect(calculatorDisplay).toMatchSnapshot();
  });
});
