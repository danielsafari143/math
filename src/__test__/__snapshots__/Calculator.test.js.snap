// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirming the display of calculator calculator displays 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculate"
      >
        <h1>
          Let's do some math!
        </h1>
        <div
          class="calc"
        >
          <input
            class="inpute"
            value=""
          />
          <input
            class="signs gray"
            type="button"
            value="A/C"
          />
          <input
            class="signs gray"
            type="button"
            value="+/-"
          />
          <input
            class="signs gray"
            type="button"
            value="%"
          />
          <input
            class="signs orange"
            type="button"
            value="รท"
          />
          <input
            class="signs gray"
            type="button"
            value="7"
          />
          <input
            class="signs gray"
            type="button"
            value="8"
          />
          <input
            class="signs gray"
            type="button"
            value="9"
          />
          <input
            class="signs orange"
            type="button"
            value="x"
          />
          <input
            class="signs gray"
            type="button"
            value="4"
          />
          <input
            class="signs gray"
            type="button"
            value="5"
          />
          <input
            class="signs gray"
            type="button"
            value="6"
          />
          <input
            class="signs signs orange"
            type="button"
            value="-"
          />
          <input
            class="signs gray"
            type="button"
            value="1"
          />
          <input
            class="signs gray"
            type="button"
            value="2"
          />
          <input
            class="signs gray"
            type="button"
            value="3"
          />
          <input
            class="signs orange"
            type="button"
            value="+"
          />
          <input
            class="signs egals gray"
            type="button"
            value="0"
          />
          <input
            class="signs gray"
            type="button"
            value="."
          />
          <input
            class="signs orange"
            type="button"
            value="="
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculate"
    >
      <h1>
        Let's do some math!
      </h1>
      <div
        class="calc"
      >
        <input
          class="inpute"
          value=""
        />
        <input
          class="signs gray"
          type="button"
          value="A/C"
        />
        <input
          class="signs gray"
          type="button"
          value="+/-"
        />
        <input
          class="signs gray"
          type="button"
          value="%"
        />
        <input
          class="signs orange"
          type="button"
          value="รท"
        />
        <input
          class="signs gray"
          type="button"
          value="7"
        />
        <input
          class="signs gray"
          type="button"
          value="8"
        />
        <input
          class="signs gray"
          type="button"
          value="9"
        />
        <input
          class="signs orange"
          type="button"
          value="x"
        />
        <input
          class="signs gray"
          type="button"
          value="4"
        />
        <input
          class="signs gray"
          type="button"
          value="5"
        />
        <input
          class="signs gray"
          type="button"
          value="6"
        />
        <input
          class="signs signs orange"
          type="button"
          value="-"
        />
        <input
          class="signs gray"
          type="button"
          value="1"
        />
        <input
          class="signs gray"
          type="button"
          value="2"
        />
        <input
          class="signs gray"
          type="button"
          value="3"
        />
        <input
          class="signs orange"
          type="button"
          value="+"
        />
        <input
          class="signs egals gray"
          type="button"
          value="0"
        />
        <input
          class="signs gray"
          type="button"
          value="."
        />
        <input
          class="signs orange"
          type="button"
          value="="
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
